src/main/java/net/minecraftforge/server/timings/ForgeTimings.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.timings;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * ForgeTimings aggregates timings data collected by {@link net.minecraftforge.server.ForgeTimeTracker} for an Object
+ * and performs operations for interpretation of the data.
+ *
+ * @param <T>
+ */
+public class ForgeTimings<T>
+{
+
+    private WeakReference<T> object;
+
+    private int[] rawTimingData;
+
+    public ForgeTimings(T object, int[] rawTimingData)
+    {
+        this.object = new WeakReference<T>(object);
+        this.rawTimingData = rawTimingData;
+    }
+
+    /**
+     * Retrieves the object that the timings are for
+     *
+     * @return The object
+     */
+    public WeakReference<T> getObject()
+    {
+        return object;
+    }
+
+    /**
+     * Returns the raw data collected by the timings tracker
+     *
+     * @return An array of the time it took (in nanoseconds) for the object to update
+     */
+    public int[] getRawTimingData()
```

Comment:
Do we expose this? Once this is exposed, it can be modified because of the nature of arrays.